services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events.py

```diff
@@ -185,7 +185,7 @@ async def action(cls, app: FastAPI, scheduler_data: SchedulerData) -> None:
 
         # Each time a new dynamic-sidecar service is created
         # generate a new `run_id` to avoid resource collisions
-        scheduler_data.run_id = uuid4()
+        scheduler_data.run_id = f"{arrow.utcnow().int_timestamp}"
```

Comment:
in a scenario where you have multiple director-v2, and many user starting services at the same time.
What is the likeliness that the same run_id will be created?
Where do you use that run_id, is there any potential for a collision somewhere? (is this used for creating a folder?) What if the created pods end up on the same machine?
It is actually quite unclear what this run_id is used for. Maybe you could add something in the description of the field...
Is this combined with the node UUID?

Reply:
> in a scenario where you have multiple director-v2, and many user starting services at the same time.
> What is the likeliness that the same run_id will be created?
If you call this API call in parallel, one of the two will fail. you cannot start multiple of them via the scheduler.
The parallel implementation will have to keep this requirement into consideration when multiple requests to start a dynamic services are run in parallel with the same parameters.
Currently, only one of them will succeed, the others will receive and error.
Also you cannot have multiple dynamic sidecars nor tracked dynamic services inside the director-v2.

> Where do you use that run_id, is there any potential for a collision somewhere? (is this used for creating a folder?) What if the created pods end up on the same machine?
It's used in the volume's name. There is no potential of collision, since you cannot start two sidecars with the same name.
Also the agent picks this value up and uses it as a folder name in s3. By the time this is store in s3, the value is guaranteed to be unique. This because you cannot have two sidecars with the same name in the system.

> It is actually quite unclear what this run_id is used for. Maybe you could add something in the description of the field...
I will add a description and see if it's clear now.
> Is this combined with the node UUID?
It has nothing to do with the node's UUID, let me know if the description is sufficient.

Reply:
Please note that the current version of the `RunID` is an extension of the previous one and no longer contains only the timestamp.